### PR TITLE
fix: prewitness deposits without delaying for a block

### PR DESCRIPTION
**Pull request by kylezs**
# Pull Request

## Checklist

Please conduct a thorough self-review before opening the PR.

- [ ] I am confident that the code works.
- [ ] I have updated documentation where appropriate.

## Summary

Means we won't delay the prewitnessing of a particular. Previously we would wait for the *next* external chain block before prewitnessing.

This release would have to be done runtime upgrade first and then CFE upgrades.

### CODE REVIEW

File: state-chain/pallets/cf-ingress-egress/src/lib.rs [hunk 1]
```diff
@@ -1752,7 +1752,8 @@ impl<T: Config<I>, I: 'static> Pallet<T, I> {
 	fn expiry_and_recycle_block_height(
 	) -> (TargetChainBlockNumber<T, I>, TargetChainBlockNumber<T, I>, TargetChainBlockNumber<T, I>)
 	{
-		let current_height = T::ChainTracking::get_block_height();
+		let current_height =
+			T::ChainTracking::get_block_height() + <T::TargetChain as Chain>::WITNESS_PERIOD;
```
Comment (martin-chainflip) [inline state-chain/pallets/cf-ingress-egress/src/lib.rs:1756]: why this change?

**Review by AlastairHolmes — APPROVED** [review]

**Review by martin-chainflip — APPROVED** [review]
